src/LedgerId.js

```diff
@@ -0,0 +1,134 @@
+import * as hex from "./encoding/hex.js";
+
+export default class LedgerId {
+    /**
+     * @hideconstructor
+     * @internal
+     * @param {Uint8Array} ledgerId
+     */
+    constructor(ledgerId) {
+        /**
+         * @readonly
+         * @type {Uint8Array}
+         */
+        this._ledgerId = ledgerId;
+
+        Object.freeze(this);
+    }
+
+    /**
+     * @param {string} ledgerId
+     * @returns {LedgerId}
+     */
+    static fromString(ledgerId) {
+        switch (ledgerId) {
+            case NETNAMES[0]:
+            case "0":
+                return LedgerId.MAINNET;
+            case NETNAMES[1]:
+            case "1":
+                return LedgerId.TESTNET;
+            case NETNAMES[2]:
+            case "2":
+                return LedgerId.PREVIEWNET;
+            default: {
+                let ledgerIdDecoded = hex.decode(ledgerId);
+                if (ledgerIdDecoded.length == 0 && ledgerId.length != 0) {
+                    throw new Error("Default reached for fromString");
+                } else {
+                    return new LedgerId(ledgerIdDecoded);
+                }
+            }
+        }
+    }
+
+    /**
+     * If the ledger ID is a known value such as `[0]`, `[1]`, `[2]` this method
+     * will instead return "mainnet", "testnet", or "previewnet", otherwise it will
+     * hex encode the bytes.
+     *
+     * @returns {string}
+     */
+    toString() {
+        if (this._ledgerId.length > 1) {
```

Comment:
I believe this is supposed to be `this._ledgerId.length === 0` since that's the only reason `this._ledgerId[0]` return `undefined`